scripts/kernel_builder.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+#
+#  Copyright (C) 2017 Intel Corporation
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+set -e
+
+if [ -n "$DEBUG" ] ; then
+	set -x
+fi
+
+KERNEL_REPO="https://github.com/clearcontainers/linux.git"
+tag_version=""
+cc_defconfig="arch/x86/configs/clear_containers_defconfig"
+
+kernel_path=$(pwd)/linux
+
+die() {
+	echo >&2 -e "\e[1mERROR\e[0m: $*"
+	exit 1
+}
+
+info() {
+	echo -e "\e[1mINFO\e[0m: $*"
+}
+
+usage() { 
+	echo "Usage: $0 <subcommand>" 1>&2;
+	echo "Options:" 1>&2;
+	echo "-h             : show this help" 1>&2;
+	echo "-k <git-repo>  : git repository to pull linux kernel" 1>&2;
+	echo "-t <tag>   : repository tag to checkout" 1>&2;
+	echo "" 1>&2;
+	echo "Subcommands:" 1>&2;
+	echo "prepare : clone linux source code and patch it with clear-containers configuration" 1>&2;
+	echo "build   : build a kernel configured by prepare subcommand" 1>&2;
+	exit 1; 
+}                                                                                                 
+
+get_last_version() {
+	#Clear Containers kernel release is tagged with \-\d+.container and is incremental
+local cc_release=$(git ls-remote --tags 2>/dev/null | grep -oP '\-\d+\.container'  | grep -oP '\d+' | sort -n | tail -1 ) 
```

Comment:
All these `git` lines are quite long so I'd be inclined to fold them with some `\`'s.